src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -344,6 +346,9 @@ void
 CreateDistributedTable(Oid relationId, Var *distributionColumn, char distributionMethod,
 					   int shardCount, char *colocateWithTableName, bool viaDeprecatedAPI)
 {
+	char replicationModel = DecideReplicationModel(distributionMethod,
```

Comment:
`DecideReplicationModel ()` ignores Citus local tables and so I was confused that we rely on its return value.

I guess the proper way would be to pass `replicationModel` parameter to `CreateDistributedTable()`. Otherwise, it is not possible to distinguish reference/citus local here. That seems like a large change.

We should probably avoid relying on `DecideReplicationModel` unless we can fix it easily (and document that we ignore those):
```C
--- a/src/backend/distributed/commands/create_distributed_table.c
+++ b/src/backend/distributed/commands/create_distributed_table.c
@@ -375,9 +375,12 @@ CreateDistributedTable(Oid relationId, Var *distributionColumn, char distributio
         * that ALTER TABLE hook does the necessary job, which means converting
         * local tables to citus local tables to properly support such foreign
         * keys.
+        *
+        * This function does not expect to create Citus local table, so we blindly
+        * create reference table when the method is DISTRIBUTE_BY_NONE.
+        *
         */
-       else if (IsCitusTableTypeInternal(distributionMethod, replicationModel,
-                                                                         REFERENCE_TABLE) &&
+       else if (distributionMethod == DISTRIBUTE_BY_NONE &&
                         ShouldEnableLocalReferenceForeignKeys() &&
                         HasForeignKeyWithLocalTable(relationId))
        {

--- a/src/backend/distributed/commands/create_distributed_table.c
+++ b/src/backend/distributed/commands/create_distributed_table.c
@@ -461,6 +461,10 @@ CreateDistributedTable(Oid relationId, Var *distributionColumn, char distributio
        }
        else if (distributionMethod == DISTRIBUTE_BY_NONE)
        {
+               /*
+                * This function does not expect to create Citus local table, so we blindly
+                * create reference table when the method is DISTRIBUTE_BY_NONE.
+                *  */
                CreateReferenceTableShard(relationId);
        }
```

Reply:
(and move back `char replicationModel = DecideReplicationModel(distributionMetho` ) call into old place